### PR TITLE
3397: Fixed missing interest period

### PR DESCRIPTION
**Pull request by b14sha**
https://platform.dandigbib.org/issues/3397

### CODE REVIEW

File: modules/fbs/fbs.info [hunk 1]
```diff
@@ -10,16 +10,27 @@ dependencies[] = ding_reservation
 dependencies[] = ding_session_cache
 dependencies[] = ding_user
 dependencies[] = entity
+dependencies[] = fbs
```
Comment (cableman) [inline modules/fbs/fbs.info]: It might raise some issues when FBS is dependent on FBS?

File: modules/fbs/fbs.info [hunk 2]
```diff
@@ -48,11 +60,4 @@ files[] = includes/classes/FBSLogDrupal.php
...
 files[] = includes/classes/FBSMisconfiguredException.php
-mtime = 1499431707
-# Features will mark any module that implements hook_field_info() as required.
-# This causes the module to be enabled automatically during site-install, which
-# in the case of a Ding-provider is a problem.
-# To prevent "required = 1" line below has been commented out. When a future
-# feature-export reintroduce the line make sure to ignore it and keep this
-# block of comments and the out-commented line intact.
-# required = 1
+required = 1
```
Comment (cableman) [inline modules/fbs/fbs.info]: Please **_read_** the comment and remove this required line.

File: modules/fbs/includes/fbs.reservation.inc [hunk 3]
```diff
@@ -326,10 +326,24 @@ function fbs_reservation_branch_name($branch_id) {
...
+  // Use a metadata wrapper to access the data.
+  $wrapper = entity_metadata_wrapper('profile2', $profile);
+
+  // Get values.
+  $interest_period = $wrapper->field_fbs_interest_period->value();
+
+  if (empty($interest_period)) {
+    $interest_period = variable_get('fbs_default_interest_period', 180);
+  }
+
```
Comment (cableman) [inline modules/fbs/includes/fbs.reservation.inc]: It seams that the code above is during the same as fbs_reservation_default_interest_period() so could that be reused?

File: modules/fbs/includes/fbs.reservation.inc [hunk 4]
```diff
@@ -377,11 +395,26 @@ function fbs_reservation_options_submit($account, $values) {
...
+    $profile = ding_user_provider_profile($user);
+  }
+
+  // Get interest period from the profile.
+  $wrapper = entity_metadata_wrapper('profile2', $profile);
+  $value = $wrapper->field_fbs_interest_period->value();
+
+  // If no period is selected, try getting default value.
+  if (is_null($value) || empty($value)) {
+    $value = variable_get('alma_default_interest_period', 180);
```
Comment (cableman) [inline modules/fbs/includes/fbs.reservation.inc]: I this you meant to use FBS and not ALMA her?

File: modules/fbs/includes/fbs.reservation.inc [hunk 5]
```diff
@@ -326,10 +326,24 @@ function fbs_reservation_branch_name($branch_id) {
...
+  // Use a metadata wrapper to access the data.
+  $wrapper = entity_metadata_wrapper('profile2', $profile);
+
+  // Get values.
+  $interest_period = $wrapper->field_fbs_interest_period->value();
+
+  if (empty($interest_period)) {
+    $interest_period = fbs_reservation_default_interest_period();
+  }
+
```
Comment (cableman) [inline modules/fbs/includes/fbs.reservation.inc]: The lines above that looks in the profile for the periode, looks like do the same "fbs_reservation_default_interest_period($account) would do before returning the default periode.

So why do this above when the function do the same?

Comment (b14sha) [reply modules/fbs/includes/fbs.reservation.inc]: Good point...

File: modules/fbs/includes/fbs.reservation.inc [hunk 6]
```diff
@@ -326,10 +326,20 @@ function fbs_reservation_branch_name($branch_id) {
...
  */
 function fbs_reservation_default_options($account) {
+  // Load profile2 profile.
+  $profile = ding_user_provider_profile($account);
+
+  // Use a metadata wrapper to access the data.
+  $wrapper = entity_metadata_wrapper('profile2', $profile);
+
+  // Get values.
+  $interest_period = fbs_reservation_default_interest_period($profile);
```
Comment (cableman) [inline modules/fbs/includes/fbs.reservation.inc]: Change the function to take in a $account and only load the profile inside this function.

File: modules/fbs/includes/fbs.reservation.inc [hunk 7]
```diff
@@ -357,9 +361,13 @@ function fbs_reservation_options($account, $default_branch = NULL) {
     $preferred_branch = $default_branch;
   }
 
+  $profile = ding_user_provider_profile($account);
```
Comment (cableman) [inline modules/fbs/includes/fbs.reservation.inc]: So close :-)

Why load the profile here, when the $profile is only used inside fbs_reservation_default_interest_period();

So why not: 

`$interest_period = fbs_reservation_default_interest_period($account);`

File: modules/fbs/includes/fbs.reservation.inc [hunk 8]
```diff
@@ -377,11 +385,26 @@ function fbs_reservation_options_submit($account, $values) {
...
+ * Implements hook_default_interest_period().
  *
- * @return int
- *   Default interest period for user.
+ * Tries to find the users default interest period based on the users profile2
+ * provider profile.
  */
-function fbs_reservation_default_interest_period() {
-  return variable_get('fbs_default_interest_period', 180);
+function fbs_reservation_default_interest_period($profile = NULL) {
```
Comment (cableman) [inline modules/fbs/includes/fbs.reservation.inc]: Then this function could be changed to:

```
function fbs_reservation_default_interest_period($account = NULL) {
  $account = is_null($account) ? $GLOBALS['user'] : $account;
  $profile = ding_user_provider_profile($user);

  // Get interest period from the profile.
  $wrapper = entity_metadata_wrapper('profile2', $profile);
  $value = $wrapper->field_fbs_interest_period->value();

  // If no period is selected, try getting default value.
  if (is_null($value) || empty($value)) {
    $value = variable_get('fbs_default_interest_period', 180);
  }
}
```
Or one could use `user_uid_optional_load` if the whole user object was required and not `$GLOBALS['user']`

File: modules/fbs/includes/fbs.reservation.inc (hunk 7 above)
Comment (kasperg) [reply modules/fbs/includes/fbs.reservation.inc]: I agree. Passing the profile to `fbs_reservation_default_interest_period` does not make any sense when the function expects an account as the argument and will load the profile based on the passed account (the same as this line does) on L395.

**Pull request merged**